sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -128,26 +147,32 @@ public CompletableResultCode forceFlush() {
     private final long scheduleDelayNanos;
     private final int maxExportBatchSize;
     private final long exporterTimeoutNanos;
-
     private long nextExportTime;
-
-    private final BlockingQueue<ReadableSpan> queue;
-
+    private final MpscArrayQueue<ReadableSpan> queue;
+    private final AtomicLong addedSpansCounter = new AtomicLong(0);
     private final AtomicReference<CompletableResultCode> flushRequested = new AtomicReference<>();
     private volatile boolean continueWork = true;
-    private final ArrayList<SpanData> batch;
+    private final Collection<SpanData> batch;
+    private final ReentrantLock lock;
+    public AtomicLong droppedSpansCounter = new AtomicLong(0);
+    public AtomicLong exportedSpansCounter = new AtomicLong(0);
```

Comment:
This is a temporary add to show the benchmark results as I couldn't get the metric to work correctly.